### PR TITLE
Make metadata retry configurable

### PR DESCRIPTION
**Pull request by pavolloffay**

### CODE REVIEW

File: exporter/kafkaexporter/kafkajaegerexporterimpl.go [hunk 1]
```diff
@@ -34,18 +34,24 @@ func newTraceExporter(o Options) (*traceExporter, error) {
...
 	return newTraceExporterWithClient(o, p), nil
 }
 
-func newAsyncProducer(brokerList []string) (sarama.AsyncProducer, error) {
+func newAsyncProducer(brokerList []string, metadataRetryMax int, metadataRetryBackOff time.Duration) (sarama.AsyncProducer, error) {
 	config := sarama.NewConfig()
 	config.Producer.RequiredAcks = sarama.WaitForLocal       // Only wait for the leader to ack
 	config.Producer.Compression = sarama.CompressionSnappy   // Compress messages
 	config.Producer.Flush.Frequency = 500 * time.Millisecond // Flush batches every 500ms
+	if metadataRetryBackOff > 0 {
```
Comment (pavolloffay) [inline exporter/kafkaexporter/kafkajaegerexporterimpl.go:49]: This is a workaround to make it use the default when no specified. I wasn't sure if the `mapstructure` accepts pointers for primitive types.

File: exporter/kafkaexporter/kafkajaegerexporterimpl.go [hunk 2]
```diff
@@ -34,18 +34,24 @@ func newTraceExporter(o Options) (*traceExporter, error) {
...
+		config.Metadata.Retry.Backoff = metadataRetryBackOff
+	}
+	if config.Metadata.Retry.Max > 0 {
```
Comment (buchi-busireddy) [inline exporter/kafkaexporter/kafkajaegerexporterimpl.go:52]: @pavolloffay This condition should be `metadataRetryMax > 0`.

Comment (pavolloffay) [reply exporter/kafkaexporter/kafkajaegerexporterimpl.go]: +1 my bad I was doing this too quickly.

Comment (pavolloffay) [reply exporter/kafkaexporter/kafkajaegerexporterimpl.go]: fixed

**Pull request merged**